torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -194,8 +194,10 @@ def __init__(
 
         # Make sure all parameters are sharded.
         for n, p in self.named_parameters():
-            if not hasattr(p, "_is_sharded"):
-                raise RuntimeError(f"found unsharded parameter: {n} ; {p.size()}")
+            if not isinstance(p, FlatParameter):
```

Comment:
let us still check 'if not hasattr(p, "_is_sharded")', which is more strict to require parameters are sharded

Reply:
Why this is more strict? `hasattr(p, "_is_sharded")` only test whether the `_is_sharded` member variable exists or not. `_is_sharded` can be False if `world_size` is 1 (or some other cases in the future). IMHO,  testing if the parameter is `FlatParameter` or not is more appropriate as in the below code we use it as it is a `FlatParameter` (e.g., assuming it has some APIs/member variables like `shard_metadata`).